src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,204 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public Node<T> getHead() {
+        return head;
+    }
+
+    public void setHead(Node<T> head) {
+        this.head = head;
+    }
+
+    public Node<T> getTail() {
+        return tail;
+    }
+
+    public void setTail(Node<T> tail) {
+        this.tail = tail;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        if (isEmpty()) {
+            node = new Node<>(null, value, null);
+            setHead(node);
+            setTail(node);
+        } else {
+            node = new Node<>(tail, value, null);
+            tail.setNext(node);
+            setTail(node);
+        }
+        setTail(node);
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " out of range " + size());
+        }
+        Node<T> insert;
+        if (index == 0 && getHead() != null) {
+            insert = new Node<>(null, value, getHead());
+            getHead().setPrev(insert);
+            setHead(insert);
+            size++;
+        } else if (getHead() == null || index == size()) {
+            add(value);
+        } else {
+            Node<T> old = getNode(index);
+            insert = new Node<>(old.getPrev(), value, old);
+            old.getPrev().setNext(insert);
+            old.setPrev(insert);
+            size++;
+        }
```

Comment:
rewrite it easier
if (index == size) call method add(T value);
otherwise:
1. check Index
2. get next node by index
3. get prev node before node that you get by index
4. create newNode using the constructor with next, prev and value
5. if prev == null set newNode as head
6. else link prev.next to newNode
7. after if else link next.prev to new node
8. increment size